src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -0,0 +1,468 @@
+    assertEffectiveLocation(item2, holdingPermId, holdingTempId, itemPermId, itemTempId);
+
+    JsonObject holding2 = holdingsClient.getById(holdingsRecordId).getJson();
+    setPermanentTemporaryLocation(holding2, newHoldingPermId, newHoldingTempId);
+    holdingsClient.replace(holdingsRecordId, holding2);
+    JsonObject item3 = itemsClient.getById(itemId).getJson();
+    assertEffectiveLocation(item3, newHoldingPermId, newHoldingTempId, itemPermId, itemTempId);
+  }
+
+  @Test
+  public void canSearchItemByEffectiveLocation() throws Exception {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+
+    UUID holdingsWithPermLocation = createInstanceAndHolding(mainLibraryLocationId);
+    UUID holdingsWithTempLocation = createInstanceAndHolding(mainLibraryLocationId, annexLibraryLocationId);
+
+    Item itemWithHoldingPermLocation = buildItem(holdingsWithPermLocation, null, null);
+    Item itemWithHoldingTempLocation = buildItem(holdingsWithTempLocation, null, null);
+    Item itemWithTempLocation = buildItem(holdingsWithPermLocation, onlineLocationId, null);
+    Item itemWithPermLocation = buildItem(holdingsWithTempLocation, null, secondFloorLocationId);
+    Item itemWithAllLocation = buildItem(holdingsWithTempLocation, secondFloorLocationId, onlineLocationId);
+
+    Item[] itemsToCreate = {itemWithHoldingPermLocation, itemWithHoldingTempLocation,
+      itemWithTempLocation, itemWithPermLocation, itemWithAllLocation};
+
+    for (Item item : itemsToCreate) {
+      IndividualResource createdItem = createItem(item);
+      assertTrue(createdItem.getJson().containsKey("effectiveLocationId"));
+    }
+
+    Items mainLibraryItems = findItems("effectiveLocationId=" + mainLibraryLocationId);
+    Items annexLibraryItems = findItems("effectiveLocationId=" + annexLibraryLocationId);
+    Items onlineLibraryItems = findItems("effectiveLocationId=" + onlineLocationId);
+    Items secondFloorLibraryItems = findItems("effectiveLocationId=" + secondFloorLocationId);
+
+    assertEquals(1, mainLibraryItems.getTotalRecords().intValue());
+    assertThat(mainLibraryItems.getItems().get(0).getId(), is(itemWithHoldingPermLocation.getId()));
+
+    assertEquals(1, annexLibraryItems.getTotalRecords().intValue());
+    assertThat(annexLibraryItems.getItems().get(0).getId(), is(itemWithHoldingTempLocation.getId()));
+
+    assertEquals(2, onlineLibraryItems.getTotalRecords().intValue());
+
+    assertThat(onlineLibraryItems.getItems()
+        .stream()
+        .map(Item::getId)
+        .collect(Collectors.toList()),
+      hasItems(itemWithTempLocation.getId(), itemWithAllLocation.getId()));
+
+    assertEquals(1, secondFloorLibraryItems.getTotalRecords().intValue());
+    assertThat(secondFloorLibraryItems.getItems().get(0).getId(), is(itemWithPermLocation.getId()));
+  }
+
+  @Test
+  public void responseContainsAllRequiredHeaders() throws Exception {
```

Comment:
Why do we need this test?

Reply:
@marcjohnson-kint, this test were created to cover post and then re-read item logic but since we are making changes to RMB, then it is redundant.

Reply:
@bohdan-suprun Thanks for the explanation